drift_dev/lib/src/writer/database_writer.dart

```diff
@@ -117,9 +117,7 @@ class DatabaseWriter {
     }
 
     // Write implementation for query methods
-    for (final query in db.queries ?? const <Never>[]) {
-      QueryWriter(query, dbScope.child()).write();
-    }
+    db.queries?.forEach((query) => QueryWriter(dbScope.child()).write(query));
```

Comment:
I mostly prefer `for` loops when the body is not a single function call (which [Effective Dart](https://dart.dev/guides/language/effective-dart/usage#avoid-using-iterableforeach-with-a-function-literal) recommends as well).

With your restructuring, we could now do

```dart
final queryWriter = QueryWriter(dbScope.child());
db.queries?.forEach(queryWriter.write);
```

Reply:
The QueryWriter still has the `final StringBuffer _buffer;` field as long as it is fine to use the same buffer for all queries that should work